scala-thoughts.md

```diff
@@ -0,0 +1,29 @@
+
+###What is/are your least favorite language design choice(s)? Why? And why do you think the designers made that / those choice(s)?
+There were a lot of things that confused / irritated me when working with Scala. One thing that was really confusing is that functions can be called without parenthesis. While this is nice, it was initially upsetting because I couldn't call certain functions with parenthesis even if I wanted to.
+
+I also found it confusing that Scala supports mutable variables and data structures if it is supposed to be a functional language. It probably would have been easier to solve some of the problems if I was allowed to use those features, especially since I have a lot of experience in declarative programming.
+
+Yet another one of Scala's strange features in the ability to call methods without having to use periods and parenthesis. Again, while this can look nice and be handy in certain situations, it makes it easier to make mistakes and could make it harder to tell what's going on. This confusion was further exacerbated when it seemed like match had to use this kind of notation (it wasn't working and one of the changes I made to try and fix it was to remove the period). Like my issue with the parenthesis and function calls, I would have prefered to be able to use a more familiar notation.
+
+I also missed return statements in Scala. It makes it really hard to return a value early on in a function as the return value is only derived from the last expression. While this wasn't so much of a problem in the practice problems, I worry that it would be hard to solve a lot of big problems with the language (as in stuff people would work on in industry).
+
+###What Scala features would you like to learn more about?
+There seems to be a lot of handy functionality in Scala that isn't always very well documented. For example, it took me a while to find out that you could use conditional expressions in case statements, which is very handy. There are also a ton of methods in Scala (many of which I did find and use in my solutions), but I still missed quite a few handy ones (like min and max for Lists). I wish it was easier to find good resources for Scala that documented all these methds and functions in one place rather than having to jump between a dozen different websites hoping that one will list a feature I hope / expect to exist in Scala.
```

Comment:
Also agreed with this... it seems like a lot of people have tried to write good introductory material for people trying to figure out how Scala is supposed to work and they've more or less just all failed to organize the information effectively.